src/main/java/io/nats/client/impl/Headers.java

```diff
@@ -337,22 +337,25 @@ public int serializedLength() {
 	 */
 	public byte[] getSerialized() {
 		if (serialized == null) {
-			ByteArrayBuilder bab = new ByteArrayBuilder(dataLength + NON_DATA_BYTES)
-					.append(VERSION_BYTES_PLUS_CRLF);
-			for (String key : valuesMap.keySet()) {
-				for (String value : valuesMap.get(key)) {
-					bab.append(key);
-					bab.append(COLON_BYTES);
-					bab.append(value);
-					bab.append(CRLF_BYTES);
-				}
-			}
-			bab.append(CRLF_BYTES);
-			serialized = bab.toByteArray();
+			serialized = appendSerialized(new ByteArrayBuilder(dataLength + NON_DATA_BYTES)).toByteArray();
 		}
 		return serialized;
 	}
 
+	public ByteArrayBuilder appendSerialized(ByteArrayBuilder bab) {
```

Comment:
I wonder if it'd further help to keep an instance of the ByteArrayBuilder in ThreadLocal to prevent creating a new object every call?  wdyt?

Reply:
Per conversation, it looks like this is reused.